app/models/ems_refresh/save_inventory_network.rb

```diff
@@ -212,24 +211,21 @@ def save_network_ports_inventory(ems, hashes, target = nil)
       h[:security_groups] = h.fetch_path(:security_groups).map { |x| x[:_object] } if h.fetch_path(:security_groups, 0, :_object)
     end
 
-    save_inventory_multi(ems.network_ports, hashes, deletes, [:ems_ref])
+    save_inventory_multi(ems.network_ports, hashes, deletes, [:ems_ref], :cloud_subnet_network_ports)
 
     store_ids_for_new_records(ems.network_ports, hashes, :ems_ref)
   end
 
-  def link_floating_ips_to_network_ports(hashes)
-    # Association of floating_ip to network_port. For backwards compatibility, we are keeping relation to Vm, even
-    # when it's redundant, since we can get vm through network_port.
-    hashes.each do |hash|
-      network_port = hash.fetch_path(:network_port, :_object)
-      # It's important we assign also blank network port, marking floating_ip association has been deleted
-
-      # TODO(lsmola) delete the check when we are not supporting nova network and grizzly, this condition
-      # prevents from deleting vm <-> floating_ip association for nova networking
-      next unless hash.key?(:cloud_network) # only neutron has cloud_network associated
+  def save_cloud_subnet_network_ports_inventory(network_port, hashes)
+    deletes = network_port.cloud_subnet_network_ports(true).dup
 
-      hash[:_object].update_attributes(:network_port => network_port)
+    hashes.each do |h|
+      %i(cloud_subnet).each do |relation|
+        h[relation] = h.fetch_path(relation, :_object) if h.fetch_path(relation, :_object)
+      end
```

Comment:
this looks like a good candidate for refactoring out the `hashes.each` block into a private function with a descriptive name :)
But I guess this is not the scope of this PR